pkg/subctl/cmd/diagnose/firewall_tunnel.go

```diff
@@ -90,87 +83,11 @@ func validateTunnelConfigAcrossClusters(localCfg, remoteCfg *rest.Config) bool {
-		return true
-	}
-
-	localEndpoint := getLocalEndpointResource(localCluster, status)
-	if localEndpoint == nil {
-		return false
-	}
-
-	gwNodeName := getActiveGatewayNodeName(localCluster, localEndpoint.Spec.Hostname, status)
-	if gwNodeName == "" {
-		return false
-	}
-
-	tunnelPort, ok := getTunnelPort(localCluster.Submariner, localEndpoint, status)
-	if !ok {
-		return false
-	}
-
-	clientMessage := string(uuid.NewUUID())[0:8]
-	podCommand := fmt.Sprintf("timeout %d tcpdump -ln -Q in -A -s 100 -i any udp and dst port %d | grep '%s'",
-		validationTimeout, tunnelPort, clientMessage)
-
-	sPod, err := spawnSnifferPodOnNode(localCluster.KubeClient, gwNodeName, podNamespace, podCommand)
-	if err != nil {
-		status.EndWithFailure("Error spawning the sniffer pod on the Gateway node: %v", err)
-		return false
-	}
-
-	defer sPod.Delete()
-
-	gatewayPodIP := getGatewayIP(remoteCluster, localCluster.Name, status)
-	if gatewayPodIP == "" {
-		status.EndWithFailure("Error retrieving the gateway IP of cluster %q", localCluster.Name)
-		return false
-	}
-
-	podCommand = fmt.Sprintf("for x in $(seq 1000); do echo %s; done | for i in $(seq 5);"+
-		" do timeout 2 nc -n -p %s -u %s %d; done", clientMessage, clientSourcePort, gatewayPodIP, tunnelPort)
-
-	// Spawn the pod on the nonGateway node. If we spawn the pod on Gateway node, the tunnel process can
-	// sometimes drop the udp traffic from client pod until the tunnels are properly setup.
-	cPod, err := spawnClientPodOnNonGatewayNode(remoteCluster.KubeClient, podNamespace, podCommand)
-	if err != nil {
-		status.EndWithFailure("Error spawning the client pod on non-Gateway node of cluster %q: %v",
-			remoteCluster.Name, err)
-		return false
-	}
-
-	defer cPod.Delete()
-
-	if err = cPod.AwaitCompletion(); err != nil {
-		status.EndWithFailure("Error waiting for the client pod to finish its execution: %v", err)
-		return false
-	}
-
-	if err = sPod.AwaitCompletion(); err != nil {
-		status.EndWithFailure("Error waiting for the sniffer pod to finish its execution: %v", err)
-		return false
-	}
-
-	if verboseOutput {
-		status.QueueSuccessMessage("tcpdump output from sniffer pod on Gateway node")
-		status.QueueSuccessMessage(sPod.PodOutput)
-	}
-
-	if !strings.Contains(sPod.PodOutput, clientMessage) {
-		status.EndWithFailure("The tcpdump output from the sniffer pod does not include the message"+
-			" sent from client pod. Please check that your firewall configuration allows UDP/%d traffic"+
-			" on the %q node.", tunnelPort, localEndpoint.Spec.Hostname)
-
-		return false
-	}
-
-	status.EndWithSuccess("Tunnels can be established on the gateway node")
-
-	return true
+	return diagnose.TunnelConfigAcrossClusters(clusterInfoFrom(localCluster), clusterInfoFrom(remoteCluster), diagnose.FirewallOptions{
+		ValidationTimeout: validationTimeout,
+		VerboseOutput:     verboseOutput,
+		PodNamespace:      podNamespace,
+	}, cli.NewStatus())
 }
 
 func newCluster(cfg *rest.Config) *cmd.Cluster {
```

Comment:
Use `exit.WithMessage` instead of `utils.ExitWithErrorMsg`.

PS. The comment is for L179 and L183 but I am not able to add comment there (I don't know why).

Reply:
yeah I didn't touch this code. Again it doesn't matter since this code will go away.

Reply:
Ditto as above.

Reply:
This function does not pertain to the new code under `cmd/subctl.`